ath79: add support for Buffalo WZR-300HP B0A1

majik-sheff opened the pull request:
ath79: add support for Buffalo WZR-300HP B0A1

The device is a renamed WZR-HP-G302H A1A0; it has also been sold under the name WZR-HP-G300NH2 and
possibly others though I don't have the hardware to verify this.

This commit provides a build for the B0A1 rev that reflects the exterior labeling and provides a clear installation path
so that it is not necessary to search the forums to find a mention of compatibility.

It is essentially a clone of the WZR-HP-G302H A1A0 commit and it will cleanly sysupgrade over previous installs of the
WZR-HP-G302H A1A0 and WZR-HP-G300NH2 builds without incident.  The only real difference is that I cleaned up the
flash layout a bit to take advantage of some otherwise wasted space.

Specifications:

    Atheros AR7242
    64 MB of RAM (DDR2)
    32 MB of Flash
        2x 16 MB SPI-NOR flash
    2.4 GHz 2T2R wifi
        Atheros AR9283
    5x 10/100/1000 Mbps Ethernet
        Atheros AR8316
    7x LEDs, 5x keys
        LED: 1x gpio-leds, 6x ath9k-leds
        key: 3x buttons, 2x slide switches
    UART header on PCB
        Vcc, GND, TX, RX from ethernet port side
        115200n8

Signed-off-by: Tim Thorpe <tim@tfthorpe.net>

majik-sheff commented:
I flubbed the commit and missed most of the files.  Please ignore this PR.

majik-sheff commented:
I flubbed the initial commit and missed most of the files.  I have updated the commit to reflect the full set of changes.

musashino205 commented on target/linux/ath79/image/generic.mk:
```diff
@@ -524,6 +524,20 @@ define Device/buffalo_wzr_ar7161
+  BUFFALO_HWVER := 4
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ledtrig-usbport
+  IMAGE_SIZE := 32320k
+  SUPPORTED_DEVICES += wzr-hp-g300nh2
+  SUPPORTED_DEVICES += wzr-hp-g302h-a1a0
```
```wzr-hp-g300nh2``` is ok, but g302h should be ```buffalo,wzr-hp-g302h-a1a0``` and these two lines can be combined to single line.
```SUPPORTED_DEVICES += wzr-hp-g300nh2 buffalo,wzr-hp-g302h-a1a0```

majik-sheff replied:
Updated and squashed

majik-sheff commented:
> `ath79: add support for Buffalo WXR-300HP B0A1`
> 
> It's not a "WZR-300HP B0A1"?

Yes, I fat-fingered the description as well.  It's corrected now.  Thanks for pointing it out.

majik-sheff commented:
> `ath79: add support for Buffalo WXR-300HP B0A1`
> 
> It's not a "WZR-300HP B0A1"?

Yes, it is.  I just mistyped the name when I was writing the commit message.  Thanks for pointing it out.

jneuhauser commented on target/linux/ath79/dts/ar7242_buffalo_wzr-300hp-b0a1.dts:
```diff
@@ -0,0 +1,243 @@
+			};
+
+			partition@40000 {
+				reg = <0x40000 0x50000>;
```
You partition sizes, the second reg arguments, looks wrong. It looks like you defined the partition end instead of the partition size.

majik-sheff replied:
Oops.  It should be fixed now.   Is there a master document that describes the purpose and format of these files?   I'm basically working from context clues right now.

jneuhauser commented on target/linux/ath79/dts/ar7242_buffalo_wzr-300hp-b0a1.dts:
```diff
@@ -0,0 +1,243 @@
+			};
+
+			art: partition@50000 {
+				reg = <0x50000 0x60000>;
```
Same as above

jneuhauser commented on target/linux/ath79/dts/ar7242_buffalo_wzr-300hp-b0a1.dts:
```diff
@@ -0,0 +1,243 @@
+
+			partition@60000 {
+				compatible = "denx,uimage";
+				reg = <0x60000 0x1ff0000>;
```
Same as above

jneuhauser commented on target/linux/ath79/dts/ar7242_buffalo_wzr-300hp-b0a1.dts:
```diff
@@ -0,0 +1,243 @@
+			};
+
+			partition@1ff0000 {
+				reg = <0x1ff0000 0x2000000>;
```
Same as above

majik-sheff replied:
Corrections made.

majik-sheff replied:
Corrections made.

majik-sheff committed:
ath79: add support for Buffalo WZR-300HP B0A1

The device is a renamed WZR-HP-G302H A1A0; it has also been sold under the name WZR-HP-G300NH2 and
possibly others though I don't have the hardware to verify this.

This commit provides a build for the B0A1 rev that reflects the exterior labeling and provides a clear installation path
so that it is not necessary to search the forums to find a mention of compatibility.

It is essentially a clone of the WZR-HP-G302H A1A0 commit and it will cleanly sysupgrade over previous installs of the
WZR-HP-G302H A1A0 and WZR-HP-G300NH2 builds without incident.  The only real difference is that I cleaned up the
flash layout a bit to take advantage of some otherwise wasted space.

Specifications:

    Atheros AR7242
    64 MB of RAM (DDR2)
    32 MB of Flash
        2x 16 MB SPI-NOR flash
    2.4 GHz 2T2R wifi
        Atheros AR9283
    5x 10/100/1000 Mbps Ethernet
        Atheros AR8316
    7x LEDs, 5x keys
        LED: 1x gpio-leds, 6x ath9k-leds
        key: 3x buttons, 2x slide switches
    UART header on PCB
        Vcc, GND, TX, RX from ethernet port side
        115200n8

Signed-off-by: Tim Thorpe <tim@tfthorpe.net>

hauke commented:
Please extract the common parts with the target/linux/ath79/dts/ar7242_buffalo_wzr-hp-g302h-a1a0.dts file into an own dtsi file and include it in both dts files. this way we can share common parts of the dts and do not have to duplicate them.

DragonBluep commented on target/linux/ath79/dts/ar7242_buffalo_wzr-300hp-b0a1.dts:
```diff
@@ -0,0 +1,243 @@
+	};
+
+	extosc: ref {
+		compatible = "fixed-clock";
```
Please remove this node and `&pll`. it has been moved to SoC dtsi. https://github.com/openwrt/openwrt/commit/520c90854ca73eb6c3d8feeda59766c90bdd4144

DragonBluep commented on target/linux/ath79/dts/ar7242_buffalo_wzr-300hp-b0a1.dts:
```diff
@@ -0,0 +1,243 @@
+
+	hub_port: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
```
The usb trigger node has been moved to SoC dtsi. https://github.com/openwrt/openwrt/commit/4778f6e959f5b9a6379968b600d721975238083c
